### PR TITLE
Add example of using Baggage

### PR DESCRIPTION
**Pull request by cijothomas**
Trying to address a FAQ: "Why my exporter is not showing Baggage...".

### CODE REVIEW

File: docs/trace/extending-the-sdk/README.md [hunk 1]
```diff
@@ -241,11 +241,26 @@ class MyProcessor : BaseProcessor<Activity>
 
 A demo processor is shown [here](./MyProcessor.cs).
 
+### Enriching Processor
+
+A common use case of writing custom processor is to enrich Activities with
```
Comment (cijothomas) [inline docs/trace/extending-the-sdk/README.md:246]:
```suggestion
A common use case of writing custom processor is to enrich activities with
```

**Review by reyang — APPROVED** [review]

File: docs/trace/extending-the-sdk/MyEnrichingProcessor.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+        // Enriching from Baggage.
+        // The below snippet adds every Baggage item.
+        foreach (var baggage in Baggage.GetBaggage())
+        {
+            activity.SetTag(baggage.Key, baggage.Value);
+        }
+
+        // The below snippet adds specific Baggage item.
+        var deviceTypeFromBaggage = Baggage.GetBaggage("device.type");
```
Comment (mic-max) [inline docs/trace/extending-the-sdk/MyEnrichingProcessor.cs:35]: Would this be common enough of a pattern used by developers to include some helper like 
- `Activity.SetTagFromBaggageIfPresent(string key)` extension function
- or `Baggage.SetActivityTagIfPresent(Actvity activity, string key)`

**Review by mic-max — APPROVED** [review]

File: src/OpenTelemetry.Api/README.md [hunk 3]
```diff
@@ -79,6 +79,11 @@ propagated out of proc using
 [Propagators](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/context/api-propagators.md).
 OpenTelemetry SDK ships a BaggagePropagator and enables it by default.
 
+It is important to note that `Baggage` is not automatically attached to any
+telemetry. User *can* explicitly read `Baggage` and use it to enrich metrics,
+logs and traces. An example of doing this for traces is shown
+[here](../../docs/trace/extending-the-sdk/README.md#processor).
```
Comment (utpilla) [inline src/OpenTelemetry.Api/README.md:85]: This link does not show how to enrich traces with `Baggage`.

File: docs/trace/extending-the-sdk/MyEnrichingProcessor.cs [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        activity.SetTag("mycustomTag", "myCustomTagValue");
+
+        // Enriching from Baggage.
+        // The below snippet adds every Baggage item.
+        foreach (var baggage in Baggage.GetBaggage())
+        {
+            activity.SetTag(baggage.Key, baggage.Value);
+        }
+
+        // The below snippet adds specific Baggage item.
```
Comment (utpilla) [inline docs/trace/extending-the-sdk/MyEnrichingProcessor.cs:34]: nit: Maybe be more specific: 
`// The below snippet adds specific Baggage item as an activity tag.`

File: docs/trace/extending-the-sdk/MyEnrichingProcessor.cs (hunk 2 above)
Comment (cijothomas) [reply docs/trace/extending-the-sdk/MyEnrichingProcessor.cs:35]: It'd be nice to have one, but not really required. Baggage as such is a new concept, and there is a possibility of SDK providing more automatic ways to get Baggage to be attached to other signals.

File: src/OpenTelemetry.Api/README.md (hunk 3 above)
Comment (cijothomas) [reply src/OpenTelemetry.Api/README.md:85]: I think it is okay, because # processor has the sub-heading ##enriching processor in the linked doc.

**Pull request merged**